### DISCUSSION TITLE
p.145 코드 디버깅하다 생긴 궁금증입니다.

### DISCUSSION THREAD

**Discussion by tiletocode** [fsoftwareengineer/todo-application#100] [질문과 답변] [answered]
Update Todo 구현에서 실습코드 2-38의 람다가 없는버전 코드를 디버깅돌리다가
이해가 되지 않는 부분이 생겼습니다.

![updateDebug](https://user-images.githubusercontent.com/95271801/159159899-f16d6325-6b42-4738-99bf-c2f18ee12f64.gif)

todo.setTitle(), todo.setDone()의 결과로 todo의 값만 바뀌어야 할것 같은데

original의 값까지 같이 바뀌는게 제 자바지식으로는 이해가 잘 안되어서요....

original의 값까지 같이 바뀌는 이유가 무엇인가요?

**Comment by fsoftwareengineer** [comment, marked as answer]
original이나 todo나 이름을 다르게 붙였을 뿐 같은 오브젝트를 가리키고 있습니다 :) 
`todo = original.get();` 에서 original.get()에  담긴 오브젝트도 [TodoEntity@11951]이고, todo에 담긴 오브젝트도 [TodoEntity@11951]입니다. 

예를들어서 제 북마크에 tiletocode님의 깃허브 주소를 tiletocode-github라고 저장하고, 다른분은 tiletocode님의깃허브라고 저장했다고 했을 때, tiletocode님이 새 리파지토리를 올리면 제가 tiletocode-github북마크를 타고 들어가던 다른분이 tiletocode님의깃허브 북마크를 타고 들어가던 같은 페이지를 봅니다.
이 비유를 todo에 대입해서 생각 해 볼 수 있습니다.

```
tiletocode님의 깃허브 주소 = [TodoEntity@11951
tiletocode-github = todo
tiletocode님의깃허브 = original.get()
```

레퍼런스
https://imasoftwareengineer.tistory.com/83
https://imasoftwareengineer.tistory.com/104
